src/test/java/tools/jackson/databind/ser/SerializationOrderTest.java

```diff
@@ -150,7 +150,7 @@ public BeanForStrictOrdering(@JsonProperty("c") int c, @JsonProperty("a") int a)
 
     @Test
     public void testImplicitOrderByCreator() throws Exception {
-        assertEquals("{\"c\":1,\"a\":2,\"b\":0}",
+        assertEquals("{\"a\":2,\"c\":1,\"b\":0}",
```

Comment:
Ok, this seems wrong: since Constructor is:

```
        @JsonCreator public BeanWithCreator(@JsonProperty("c") int c, @JsonProperty("a") int a)
```

order should remain "c", "a" (creator parameters), "b", given that `MapperFeature.SORT_CREATOR_PROPERTIES_FIRST` is left enabled (default state).

So something is wrong... not sure if same is true for 2.x.

Reply:
Ohhhhh. No, actually, I am wrong. Code in `POJOPropertiesCollector` says:

```
protected void _sortProperties(Map<String, POJOPropertyBuilder> props) {
....
        // Third by sorting Creator properties before other unordered properties
        // (unless strict ordering is requested)
        if ((_creatorProperties != null)
                && (!sortAlpha || _config.isEnabled(MapperFeature.SORT_CREATOR_PROPERTIES_FIRST))) {
            /* As per [databind#311], this is bit delicate; but if alphabetic ordering
             * is mandated, at least ensure creator properties are in alphabetic
             * order. Related question of creator vs non-creator is punted for now,
             * so creator properties still fully predate non-creator ones.
             */
```

that is, ALL Creator properties should be sorted before all non-Creator properties (unless explicit order dictates otherwise).

So in that sense this works as expected.

Reply:
However: this brings up another question, mentioned elsewhere: should ordering of Record properties be ordered by declaration order because it:

1. Is actually well-defined and stable
2. May well be what users expect, as the default choice

I think I better file a separate issue for that; not strictly related to this PR, but something worth considering.

Reply:
I haven't had time to look at the sorting functionality wrt creator-properties yet, but my first intuition was wondering...

    how is `record` related to creators?

But it seems like `record` constructors , by its nature, is considered creators or similar. I think I do need to (and will) learn more about records.

Thank you for the explanation tho @cowtowncoder !

Reply:
So, Records are handled quite similar to POJOs with 2.18, in that their Constructors are selected either from explicit annotation, or, if none, by selection so-called "Canonical" constructor (one that has all record fields in order).

But ordering of properties is then done separately across set of properties, not considering Creator (constructor or factory method).
(and of course actual Creator is not used for serialization anyway, nor are CreatorProperty instances -- but `POJOPropertiesCollector` has information on whether logical property has a creator-parameter binding).

I think sorting is handled by `POJOPropertiesCollector` and could keep creator-bound properties in order they were added (which is declaration order).

I hope 2.18 code, esp. in `POJOPropertiesCollector`, is easier to follow through than before refactoring. There is still some later processing in `BasicDeserializerFactory`, but more of logic is now in PPC.